Modules/_datetimemodule.c

```diff
@@ -637,17 +637,22 @@ check_date_args(int year, int month, int day)
 {
 
     if (year < MINYEAR || year > MAXYEAR) {
-        PyErr_Format(PyExc_ValueError, "year %i is out of range", year);
+        PyErr_Format(PyExc_ValueError,
+                     "year must be in %d..%d, but got %d",
+                     MINYEAR, MAXYEAR, year);
         return -1;
     }
     if (month < 1 || month > 12) {
-        PyErr_SetString(PyExc_ValueError,
-                        "month must be in 1..12");
+        PyErr_Format(PyExc_ValueError,
+                    "month must be in 1..12, but got %d",
+                    month);
         return -1;
     }
-    if (day < 1 || day > days_in_month(year, month)) {
-        PyErr_SetString(PyExc_ValueError,
-                        "day is out of range for month");
+    int dim = days_in_month(year, month);
+    if (day < 1 || day > dim) {
+        PyErr_Format(PyExc_ValueError,
+                    "day must be in 1..%d, but got %d",
+                    dim, day);
```

Comment:
Nit: this fits well within 79 chars.
Moreover: fix vertical alignment.

```suggestion
                     "day must be in 1..%d, but got %d", dim, day);
```